src/test/scala/ECommAlgorithmTest2.scala

```diff
@@ -0,0 +1,408 @@
+      (math floor (featureU5.get(9)*1000) )/1000 shouldBe -0.263
+      
+      
+      // check the first and last coordinate of the feature vector of the first item
+      (math floor (featureI1.get(0)*1000) )/1000 shouldBe 0.351
+      (math floor (featureI1.get(9)*1000) )/1000 shouldBe 0.684
+
+      // check the first and last coordinate of the feature vector of the last item
+      (math floor (featureI5.get(0)*1000) )/1000 shouldBe -0.031
+      (math floor (featureI5.get(9)*1000) )/1000 shouldBe -0.216
+      
+	}    
+
+  
+//======================================================================================================
+//======================================================================================================  
+  
+  // test of the trainDefault method
+	"ECommAlgorithm.trainDefault()" should "return correct user and item feature vectors" in {
+
+//		addToLog("\n---------- train default TEST ----------\n")     
+
+
+		val preparedData = new PreparedData(
+				users = sc.parallelize(users.toSeq),
+				items = sc.parallelize(items.toSeq),
+				viewEvents = sc.parallelize(view.toSeq),
+				buyEvents = sc.parallelize(buy.toSeq)
+		)
+    
+		// test traindefault
+		val mapPopCount = algorithm.trainDefault(preparedData)
+
+		val i1Count = mapPopCount.get(1)
+		val i2Count = mapPopCount.get(2)
+		val i3Count = mapPopCount.get(3)
+		val i4Count = mapPopCount.get(4)
+		val i5Count = mapPopCount.get(5)
+
+//		addToLog("i1Count = " + i1Count.toString() + "\n")
+//		addToLog("i2Count = " + i2Count.toString() + "\n")
+//		addToLog("i3Count = " + i3Count.toString() + "\n")
+//		addToLog("i4Count = " + i4Count.toString() + "\n")
+//		addToLog("i5Count = " + i5Count.toString() + "\n")  
+
+
+    // TODO it remains to implement the actual test
+    val expected = None
+    
+    i1Count shouldBe expected
+    i2Count shouldBe expected
+    i3Count shouldBe expected
+    i4Count shouldBe expected
+    i5Count shouldBe expected
+	}
+
+  
+//======================================================================================================
+//======================================================================================================  
+  
+  // test the genBlackList method of class algorithm
+	"ECommAlgorithm generate black list" should "return correct black list" in {
+
+   
+		val preparedData = new PreparedData(
+				users = sc.parallelize(users.toSeq),
+				items = sc.parallelize(items.toSeq),
+				viewEvents = sc.parallelize(view.toSeq),
+				buyEvents = sc.parallelize(buy.toSeq)
+				)
+
+		val mllibRatings = algorithm.genMLlibRating( data = preparedData )
+
+		val m = ALS.trainImplicit(
+				ratings = mllibRatings,
+				rank = 10,
+				iterations = 10,
+				lambda = 0.01,
+				blocks = -1,
+				alpha = 1.0,
+				seed  = 3
+		)
+
+		val userFeatures = m.userFeatures.collectAsMap.toMap
+		val F = m.productFeatures.collectAsMap.toMap
+
+    // TODO it remains to implement the actual test
```

Comment:
removed this method. this has been an attempt to test the method genBlackList of ECommAlgorithm class